tools/reach-calculator.html

```diff
@@ -0,0 +1,233 @@
+            <label><input type="radio" id="medium" name="confidenceInput" value="medium"> Medium confidence</label><br>
+            <label><input type="radio" id="high" name="confidenceInput" value="high"> High confidence</label><br>
+        </div>
+    
+        <hr>
+        <div id="results"></div>
+        <div>(Use this and the Severity Assessment Matrix to get a priority level.)</div>
+    </div>
+</div>
+
+<script>
+(function () {
+    'use strict';
+
+    // Multipliers for confidence that the breakage issue is more widespread.
+    const confidenceLevels = {
+        low: 1,
+        medium: 10,
+        high: 100
+    };
+
+    // Format results for display so they're between 0 and 100.
+    function formatResult (num) {
+        if (num === 0 || isNaN(num)) {
+            return 0;
+        } else if (num > 1) {
+            return 100;
+        } else {
+            num *= 100;
+            return num.toPrecision(3);
+        }
+    }
+
+    // Hide/display the domain rank and API usage fields.
+    function toggleTab (type) {
+        document.body.className = type;
+        if (type === 'domain') {
+            apiUsageInput.value = '';
+        } else {
+            domainRankInput.value = '';
+        }
+        doCalculation();
+    }
+
+    // Calculate breakage reach as a percentage of page loads.
+    function doCalculation () {
+        const domainRank = domainRankInput.value;
+        const apiUsage = apiUsageInput.value;
+        const reductionFactor = reductionFactorInput.value;
+        const confidence = document.querySelector('input[name="confidenceInput"]:checked').value;
+        let result = '';
+
+        // Domain popularity is calculated using a trend line formula based on rank:pageload data from https://dl.acm.org/doi/10.1145/3517745.3561418
+        // Domain rank 1 is an outlier and therefore an exception to the formula.
+        let domainPopularity = 0;
+        if (domainRank) {
+            domainPopularity = (domainRank > 1) ? (0.0687189317758425 * Math.pow(domainRank, -1.03435633987409)) : 0.17;
+        }
+
+        const domainPopularityOutput = document.getElementById('domainPopularityOutput');
+        domainPopularityOutput.innerHTML = 'Domain pageloads: ' + formatResult(domainPopularity) + '%';
+
+        if (domainRank && !reductionFactor) {
+            result = domainPopularity;
+        } else if (apiUsage && !reductionFactor) {
+            result = apiUsage / 100;
+        } else if (domainRank && reductionFactor) {
+            result = domainPopularity * (reductionFactor / 100);
+        } else if (apiUsage && reductionFactor) {
+            result = (apiUsage / 100) * (reductionFactor / 100);
+        }
+
+        result *= confidenceLevels[confidence];
+        document.getElementById('results').innerHTML = 'Platform pageloads affected: ' + formatResult(result) + '%';
```

Comment:
These `innerHTML` aren't user content but they easily could transition to being. let's use `textContent` instead looks like there's no formatting to make this difficult.

Reply:
Good idea – done.